v0.5/tools/submission/submission-checker.py

```diff
@@ -0,0 +1,238 @@
+"""
+A checker for mlperf inference submissions
+"""
+
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import collections
+import json
+import logging
+import os
+import re
+import sys
+import time
+
+# pylint: disable=missing-docstring
+
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger("main")
+
+VALID_MODELS = ["ssd-small", "ssd-large", "mobilenet", "resnet", "gnmt"]
+REQUIRED_DIRECTORIES = ["code", "results", "measurements", "systems"]
+REQUIRED_PERF_FILES = ["mlperf_log_accuracy.json", "mlperf_log_summary.txt", "mlperf_log_detail.txt"]
+REQUIRED_ACC_FILES = REQUIRED_PERF_FILES + ["accuracy.txt"]
+REQUIRED_MEASURE_FILES = ["mlperf.conf", "README.md"]
+
+
+def get_args():
+    """Parse commandline."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--input", required=True, help="submission directory")
+    args = parser.parse_args()
+    return args
+
+
+def path_to_dict(path):
+    d, f = {}, {}
+    for dirName, subdirList, fileList in os.walk(path):
+        d[dirName] = sorted(subdirList)
+        f[dirName] = sorted(fileList)
+    return d, f
+
+
+def check_accuracy(model, dir):
+    is_valid = False
+    # look for: accuracy=... or mAP=...
+    with open(os.path.join(dir, "accuracy.txt"), "r") as f:
```

Comment:
What is this `accuracy.txt`? I can't find it in submission rules: https://github.com/mlperf/policies/blob/master/submission_rules.adoc

Reply:
Its is the output of the accuracy-[coco|imagenet].py script. 
It is called from the sample submission notebook. 
I might change the code a little to actually create the file but than I need to have coco and imagenet datasets passed into the checker.

Reply:
Or change `accuracy-[coco|imagenet].py` so that they store accuracy results to disk? I am worried about calling those scripts from here because each implementation might have different flags which are passed into the accuracy scripts.

Reply:
Otherwise, we will at least have to provide users with a way to specify which flags they want to pass into accuracy scripts.

Reply:
The user need to run those scripts anyway, else he would would not know if he met accuracy.
I can add a message with some instructions.

Reply:
I changed to print instructions to run those scripts if I can't find accuracy.txt

Reply:
this plan sounds good.